packages/api/src/modules/subventions/subventions.service.ts

```diff
@@ -41,23 +44,38 @@ export class SubventionsService {
         return await Promise.any(promises).catch(() => null);
     }
     
-    private async aggregate(id: StructureIdentifiers, type: Record<StructureIdentifiersEnum, string>[StructureIdentifiersEnum]): Promise<DemandeSubvention[]> {
-        let subventions;
+    private aggregate(id: StructureIdentifiers, type: Record<StructureIdentifiersEnum, string>[StructureIdentifiersEnum]): Flux<SubventionsFlux> {
         if (type === StructureIdentifiersEnum.siret || type === StructureIdentifiersEnum.siren) {
             if (type === StructureIdentifiersEnum.siret) id = siretToSiren(id);
-            subventions = await this.aggregateByType(id, StructureIdentifiersEnum.siren)
-            if (subventions.length > 0) return subventions;
+            return this.aggregateByType(id, StructureIdentifiersEnum.siren);
         }
         else {
-            subventions = await this.aggregateByType(id, StructureIdentifiersEnum.rna)
+            return this.aggregateByType(id, StructureIdentifiersEnum.rna);
         }
-        return subventions;
     }
     
-    private async aggregateByType(id: StructureIdentifiers, type: StructureIdentifiersEnum): Promise<DemandeSubvention[]> {
+    private aggregateByType(id: StructureIdentifiers, type: StructureIdentifiersEnum): Flux<SubventionsFlux> {
         const functionName = `getDemandeSubventionBy${capitalizeFirstLetter(type)}` as "getDemandeSubventionBySiret" | "getDemandeSubventionBySiren" | "getDemandeSubventionByRna";
-        const promises = this.getDemandesSubventionsProviders().map(p =>  p[functionName](id));
-        return [...(await Promise.all(promises)).flat()] as DemandeSubvention[];
+        const subventionsFlux = new Flux<SubventionsFlux>();
+        const providers = this.getDemandesSubventionsProviders();
+        
+        let countAnswers = 0;
+        
+        providers.forEach(p => p[functionName](id).then(subventions => {
+            countAnswers++;
+    
+            subventionsFlux.push({
```

Comment:
On peut pas sauvegarder ces meta données directement dans la classe (a.k.a Object) subventionsFlux ? 
```
subventionsFlux.__meta__ = {
                     providerCalls: providers.length,
                     providerAnswers: countAnswers
                 }
```

de cette façon c'est bien une mise à jour globale et non pas seulement sur le dernier élément inséré ?

Reply:
Le souci de le passer dans subvention flux, c'est que le type flux à pour but d'être réutilisé dans d'autre cas que celui des subventions, et on n'aura pas forcements de providers.

Si on veux séparer le deux il faudrait faire deux flux, mais in fine il faudrait quand même remontée les deux informations via le sse

Reply:
Je veux bien qu'on face un point en direct tous les deux ! Pour etre sur d'avoir compris ce que tu voulais dire

Reply:
Yep faisons un point ! Je veux bien plus d'explications sur le commentaire plus haut sur les prevData et l'appel du callback avec toutes les anciennes données!